botorch/optim/optimize.py

```diff
@@ -956,15 +979,18 @@ def optimize_acqf_mixed(
                     post_processing_func=post_processing_func,
                     batch_initial_conditions=batch_initial_conditions,
                     ic_generator=ic_generator,
-                    return_best_only=True,
+                    return_best_only=False,
```

Comment:
Why not `return_best_only=return_best_only`?

Reply:
We get here all candidates and then filter later based on the `return_best_only` argument. I will add a comment.